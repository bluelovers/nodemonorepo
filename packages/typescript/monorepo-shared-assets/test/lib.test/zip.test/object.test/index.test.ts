import { zip } from '../../../../index'
const { FIRST, INNER_JOIN, OUTER_JOIN } = zip.object

const foo = {
  sharedA: 'fooA',
  sharedB: 'fooB',
  ownFooC: 'fooC',
  ownFooD: 'fooD'
}

const bar = {
  sharedA: 'barA',
  sharedB: 'barB',
  ownBarC: 'barC',
  ownBarD: 'barD'
}

describe('with KeySetProvider=FIRST', () => {
  it('with foo as first argument', () => {
    const result = zip.object(foo, bar, FIRST)
    expect(result).toMatchSnapshot()
    expect(result).toEqual(zip.object(foo, bar)) // default parameter
  })

  it('with bar as first argument', () => {
    const result = zip.object(bar, foo, FIRST)
    expect(result).toMatchSnapshot()
    expect(result).toEqual(zip.object(bar, foo)) // default parameter
  })
})

describe('with KeySetProvider=INNER_JOIN', () => {
  it('with foo as first argument', () => {
    const result = zip.object(foo, bar, INNER_JOIN)
    expect(result).toMatchSnapshot()
  })

  it('with bar as first argument', () => {
    const result = zip.object(bar, foo, INNER_JOIN)
    expect(result).toMatchSnapshot()
  })
})

describe('with KeySetProvider=OUTER_JOIN', () => {
  it('with bar as first argument', () => {
    const result = zip.object(bar, foo, OUTER_JOIN)
    expect(result).toMatchSnapshot()
  })
})
