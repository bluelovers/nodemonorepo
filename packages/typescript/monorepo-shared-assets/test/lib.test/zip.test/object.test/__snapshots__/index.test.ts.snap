// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with KeySetProvider=FIRST with bar as first argument 1`] = `
Object {
  "ownBarC": Array [
    "barC",
    undefined,
  ],
  "ownBarD": Array [
    "barD",
    undefined,
  ],
  "sharedA": Array [
    "barA",
    "fooA",
  ],
  "sharedB": Array [
    "barB",
    "fooB",
  ],
}
`;

exports[`with KeySetProvider=FIRST with foo as first argument 1`] = `
Object {
  "ownFooC": Array [
    "fooC",
    undefined,
  ],
  "ownFooD": Array [
    "fooD",
    undefined,
  ],
  "sharedA": Array [
    "fooA",
    "barA",
  ],
  "sharedB": Array [
    "fooB",
    "barB",
  ],
}
`;

exports[`with KeySetProvider=INNER_JOIN with bar as first argument 1`] = `
Object {
  "sharedA": Array [
    "barA",
    "fooA",
  ],
  "sharedB": Array [
    "barB",
    "fooB",
  ],
}
`;

exports[`with KeySetProvider=INNER_JOIN with foo as first argument 1`] = `
Object {
  "sharedA": Array [
    "fooA",
    "barA",
  ],
  "sharedB": Array [
    "fooB",
    "barB",
  ],
}
`;

exports[`with KeySetProvider=OUTER_JOIN with bar as first argument 1`] = `
Object {
  "ownBarC": Array [
    "barC",
    undefined,
  ],
  "ownBarD": Array [
    "barD",
    undefined,
  ],
  "ownFooC": Array [
    undefined,
    "fooC",
  ],
  "ownFooD": Array [
    undefined,
    "fooD",
  ],
  "sharedA": Array [
    "barA",
    "fooA",
  ],
  "sharedB": Array [
    "barB",
    "fooB",
  ],
}
`;
